src/release.rs

```diff
@@ -161,6 +174,46 @@ impl BuildConfig {
         Ok(label)
     }
 
+    fn clone_and_checkout(&self) -> Result<PathBuf, Box<dyn Error>> {
+        let repo_path = match &self.deploy_method {
+            DeployMethod::Commit(commit, user, repo_name) => {
+                let repo_dir_name = format!("{repo_name}_{user}");
+                let git_repo = format!("https://github.com/{}/{}.git", user, repo_name);
+                let repo_path = self.cluster_root_path.join(repo_dir_name);
+
+                if !repo_path.exists() {
+                    std::fs::create_dir_all(&repo_path).unwrap();
+                    let progress_bar = new_spinner_progress_bar();
+                    progress_bar.set_message(format!("{CLONE}Cloning Repo..."));
+                    match Repository::clone(&git_repo, repo_path.clone()) {
+                        Ok(_) => (),
+                        Err(e) => return Err(format!("Failed to clone {git_repo}. Is the github user and repo name correct?\nErr: {e}").into())
+                    }
+                    progress_bar.finish_and_clear();
+                    info!(
+                        "Successfully cloned repo: {git_repo} into {}",
+                        self.cluster_root_path.join("repo_name").display()
+                    );
+                }
+
+                let repo = Repository::open(repo_path.clone())?;
+                let git_commit = repo.find_commit(Oid::from_str(commit).unwrap())?;
+                repo.checkout_tree(git_commit.as_object(), None)?;
+                repo.set_head_detached(git_commit.id())?;
+                info!("Checked out commit: {commit}");
+                repo_path
```

Comment:
It doesn't need to be done in this PR, but it would be neater to *only* clone the desired commit following the steps at https://graphite.dev/guides/git-clone-specific-commit#additional-operations